extensions/amp-a4a/0.1/test/test-amp-a4a.js

```diff
@@ -1355,125 +1355,96 @@ describe('amp-a4a', () => {
         });
       });
     });
-    it('should collapse for 204 response code', () => {
-      return createIframePromise().then(fixture => {
-        setupForAdTesting(fixture);
-        adResponse.status = 204;
-        adResponse.body = null;
-        fetchMock.getOnce(
-            TEST_URL + '&__amp_source_origin=about%3Asrcdoc', () => adResponse,
-            {name: 'ad'});
-        const doc = fixture.doc;
-        const a4aElement = createA4aElement(doc);
-        const a4a = new MockA4AImpl(a4aElement);
-        a4a.buildCallback();
-        const forceCollapseSpy = sandbox.spy(a4a, 'forceCollapse');
-        const noContentUISpy = sandbox.spy();
-        const unlayoutUISpy = sandbox.spy();
-        a4a.uiHandler = {
-          applyNoContentUI: () => {noContentUISpy();},
-          applyUnlayoutUI: () => {unlayoutUISpy();},
-        };
-        sandbox.stub(a4a, 'getLayoutBox').returns({width: 123, height: 456});
-        a4a.onLayoutMeasure();
-        expect(a4a.adPromise_).to.be.ok;
-        return a4a.adPromise_.then(() => {
-          expect(forceCollapseSpy).to.be.calledOnce;
-          expect(noContentUISpy).to.be.calledOnce;
-          return a4a.layoutCallback().then(() => {
-            // should have no iframe.
-            expect(a4aElement.querySelector('iframe')).to.not.be.ok;
-            expect(onCreativeRenderSpy).to.not.be.called;
-            // call unlayout callback and verify it attempts to revert the size
-            expect(a4a.originalSlotSize_).to.deep
-                .equal({width: 123, height: 456});
-            let attemptChangeSizeResolver;
-            const attemptChangeSizePromise = new Promise(resolve => {
-              attemptChangeSizeResolver = resolve;
-            });
-            sandbox.stub(AMP.BaseElement.prototype, 'attemptChangeSize')
-                .returns(attemptChangeSizePromise);
-            a4a.unlayoutCallback();
-            expect(unlayoutUISpy).to.be.calledOnce;
-            expect(a4a.originalSlotSize_).to.be.ok;
-            attemptChangeSizeResolver();
-            return Services.timerFor(a4a.win).promise(1).then(() => {
-              expect(a4a.originalSlotSize_).to.not.be.ok;
+
+    [
+      {
+        name: '204',
+        fn: () => adResponse.status = 204,
+      },
+      {
+        name: 'empty body',
+        fn: () => adResponse.body = '',
+      },
+      {
+        name: 'no fill header',
+        fn: () => adResponse.headers['amp-ff-empty-creative'] = 'true',
+      },
```

Comment:
Should we collapse on error responses?  Are 500 etc tested somewhere else?

Reply:
Done